src/main/java/mate/academy/spring/controller/OrderController.java

```diff
@@ -33,14 +35,18 @@ public OrderController(ShoppingCartService shoppingCartService,
     }
 
     @PostMapping("/complete")
-    public OrderResponseDto completeOrder(@RequestParam Long userId) {
-        ShoppingCart cart = shoppingCartService.getByUser(userService.get(userId));
+    public OrderResponseDto completeOrder(Authentication authenticatedUser) {
+        User user = userService.findByEmail(authenticatedUser
+                .getName()).orElseThrow(NoSuchElementException::new);
+        ShoppingCart cart = shoppingCartService.getByUser(userService.get(user.getId()));
```

Comment:
Replace userService.get(user.getId()) with 'user' in the 'getOrderHistory' method, as the 'user' object is already fetched in the previous line.